sklearn/neighbors/binary_tree.pxi

```diff
@@ -1001,11 +1007,18 @@ VALID_METRIC_IDS = get_valid_metric_ids(VALID_METRICS)
 cdef class BinaryTree:
 
     cdef np.ndarray data_arr
+    cdef np.ndarray new_data_arr
+    cdef bint is_float
+
     cdef np.ndarray idx_array_arr
     cdef np.ndarray node_data_arr
     cdef np.ndarray node_bounds_arr
 
     cdef readonly DTYPE_t[:, ::1] data
+    cdef void *new_data
+    cdef int n_samples
```

Comment:
These should provide little benefit over `self.data_arr.shape[x]`, I think, as that is defined in the `pxd` as a field of `ndarray`.